### DISCUSSION TITLE
Memcached module suddenly gets installed on PHP 8

### DISCUSSION THREAD

**Discussion by driesvints** [shivammathur/setup-php#371] [Q&A / Help] [answered]
For some reason the Memcached extension is finally being installed on our PHP 8 builds for laravel/framework.

Before: https://github.com/laravel/framework/runs/1563519887?check_suite_focus=true#step:4:27
Now: https://github.com/laravel/framework/runs/1569670795#step:4:27

[This causes some tests to fail on PHP 8](https://github.com/laravel/framework/runs/1569670795#step:8:99). That's a good thing because we can identify these bugs and fix them now. However, I have no idea why suddenly the extension is being installed. https://github.com/php-memcached-dev/php-memcached doesn't has a new release tagged. Does anyone knows why this is suddenly happening? Then I can maybe try to install it locally as well in order to reproduce the failures.

@shivammathur Sorry for pinging you directly but it would be really appreciated if you had any insights here.

**Comment by shivammathur** [comment, marked as answer]
@driesvints You can always reach out directly 😀
Previously installing `memcached` on PHP 8.0 required installing `libmemcached` manually before `setup-php`.

Now GitHub Actions ubuntu environments have PHP 8.0 from `ppa:ondrej/php` pre-installed along with `php8.0-memcached` package.
Until you fix, if you want to remove `memcached`, you can specify it in extensions input prefixed with colon `:`

```yml
- name: Setup PHP
  uses: shivammathur/setup-php@v2
  with:
    php-version: '8.0'
    extensions: :memcached
```
Test workflow: https://github.com/shivammathur/test-setup-php/actions/runs/428074802/workflow

**Reply by driesvints** [reply]
Awesome, thanks! I actually don't want to remove the memcached extension because it surfaces some bugs we need to fix. I'm not mostly looking into a way of setting it up locally.

**Reply by driesvints** [reply]
I can't seem to mark your answer as the solution to this discussion. Not sure why GitHub doesn't allows me to do that.

**Reply by shivammathur** [reply]
>  looking into a way of setting it up locally.

If you use ubuntu
```
# Install PHP 8 and memcached
sudo apt-get install php8.0 php8.0-memcached -y
```
For macos
```
# Install php 8
brew tap shivammathur/php
brew install shivammathur/php/php@8.0
brew link --force --overwrite php@8.0

# Install memcached
brew tap kabel/pecl
brew install kabel/pecl/php-memcached
```

**Reply by driesvints** [reply]
Amazing, thanks! 🙂

**Comment by driesvints** [comment]
@shivammathur It seems this still doesn't entirely resolves it for me sorry. I managed to get php-memcached installed locally and when I run the test suite for Memcached in Laravel everything passes fine. But on our CI build it continuous to fail. The most annoying thing about this all is that there's no stack trace. So we have no idea where the issue is coming from.

Please see the failing build here: https://github.com/laravel/framework/pull/35647/checks?check_run_id=1571646358#step:8:99

Any help is appreciated because I've really exhausted my abilities here 😞

**Comment by shivammathur** [comment]
Mockery is calling `getName()` method on `ReflectionUnionType`. In PHP 8 `getName()` method does not exist on `ReflectionUnionType`. It does exist on `ReflectionNamedType`.

**Reply by driesvints** [reply]
Thanks! I'm a little further now and discovered that union types aren't supported in phpunit 8.x
